instance/tasks.py

```diff
@@ -339,3 +344,111 @@ def cleanup_old_betatest_users():
         users_to_delete = queryset.filter(is_active=False, profile__modified__lte=delete_cutoff)
         logger.info('Deleting %s users.', users_to_delete.count())
         users_to_delete.delete()
+
+
+class KillZombiesRunner:
+    """
+    Helper class to run `kill_zombies_periodically`
+    """
+
+    def __init__(self):
+        self.region: str = getattr(settings, "OPENSTACK_REGION", "gra")
+        self.recipient: list = [email for _, email in getattr(settings, "ADMINS", [])]
+        self.threshold: int = getattr(settings, "KILL_ZOMBIES_WARNING_THRESHOLD")
+
+    def send_email(self, subject: str, body: str) -> int:
+        """
+        Utility method for sending emails
+        """
+        if self.recipient:
+            email = EmailMessage(
+                subject, body, settings.DEFAULT_FROM_EMAIL, self.recipient
+            )
+            return email.send()
+        else:
+            logging.info("Email recipient is undefined. Skipping email.")
+            return None
+
+    def trigger_warning(self):
+        """
+        Warns when more than `threshold` VMs will be terminated
+        """
+        subject = "{} Zombie instances over current threshold ({})".format(
+            datetime.now().strftime("%Y-%m-d"), self.threshold
+        )
+        body = (
+            "The number of instances in region {} to be terminated"
+            "is over KILL_ZOMBIES_WARNING_THRESHOLD ({}).\n"
+        ).format(self.region, self.threshold)
+        logging.info(
+            "%s\nSending an email notification to: %s",
+            body, self.recipient
+        )
+        self.send_email(subject, body)
+
+    def handle_dry_run_output(self, stdout: str) -> int:
+        """
+        If there are servers to terminate, a `kill_zombies` dry run will warn:
+        "Would have terminated {} zombies" to stdout.
+        """
+        if "No servers found in region" in stdout:
+            return 0
+        try:
+            return int(stdout.split("Would have terminated")[1].split()[0])
+        except IndexError:
+            logger.info("Received unexpected input from dry run")
+            return None
```

Comment:
The code that invokes this function only treats it as an integer. So this will cause trouble there. I'd recommend using a variable `number_of_zombies_to_terminate` with `0` as its default value and returning that.

Reply:
I made it a`return None` because it would fail only if `kill_zombies` should fail. Passing a `0` instead of raising an error will allow `kill_zombies_periodically`'s `call_command` to `kill_zombies` to fail silently.
I applied the suggestion nonetheless.

Reply:
@spokerman12, if failing silently is a concern, we should notify the failure. Otherwise, it will still be undetected even though it is in the logs.

Reply:
Applied the suggestion.